File: Python Course/Software Development Module - in Python/Revision/Airline.py
"""
Airline.py
Billy Cussen
20/10/2020
"""

class myAirline:
    def __init__(self, membership, flights):
        self.membership = membership
        self.flights = flights

    def getPoints(self):
        if self.membership == "bronze":
            if self.flights <= 3:
                return "Congratulations, you earned "+str(self.flights*2)+" points"
            else:
                return "Congratulations, you earned "+str(self.flights*3)+" points"
        elif self.membership == "silver":
            if self.flights <= 3:
                return "Congratulations, you earned "+str(self.flights*4)+" points"
            else:
                return "Congratulations, you earned "+str(self.flights*5)+" points"
        elif self.membership == "gold":
            if self.flights <= 3:
                return "Congratulations, you earned "+str(self.flights*6)+" points"
            else:
                return "Congratulations, you earned "+str(self.flights*7)+" points"
        else:
            return "Somethings went Wrong here!"

File: Java/Software Development Module - Java/Revision/Airline/Airline.java
/*
Airline.java
Billy Cussen
29/07/2020
*/

public class Airline{

    //Data Members
    private String status;
    private int flights;
    private String pointsEarned;

    //Constructor
    public Airline(String status, int flights){
        this.status = status;
        this.flights = flights;
        this.pointsEarned = new String();
    }

    //Compute & Getter
    public String getPointsEarned(){

        if(this.status.equalsIgnoreCase("Silver")){
            if(this.flights <= 3){
                this.pointsEarned = "Congratulations, you get "+this.flights*2 + " points!";
            }else{
                this.pointsEarned = "Congratulations, you get "+this.flights*3 + " points!";
            }
        } else if(this.status.equalsIgnoreCase("Gold")){
            if(this.flights <= 3){
                this.pointsEarned = "Congratulations, you get "+this.flights*4 + " points!";
            }else{
                this.pointsEarned = "Congratulations, you get "+this.flights*5 + " points!";
            }
        } else{
            if(this.flights <= 3){
                this.pointsEarned = "Congratulations, you get "+ ((int)Math.pow(this.flights,2)) + " points!";
            }else{
                this.pointsEarned = "Congratulations, you get "+ ((int)Math.pow(this.flights,3)) + " points!";
            }            
        }

        return this.pointsEarned;
    }

}